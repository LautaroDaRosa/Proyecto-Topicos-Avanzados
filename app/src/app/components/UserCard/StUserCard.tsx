import styled from 'styled-components';

const StUserCard = styled.div`
  border-radius: 4px;
  display: flex;
  padding: 10px;
  transition: 0.2s background-color ease-in-out;
  text-decoration: none;

  img {
    width: 28px;
    border-radius: 50%;
    margin-right: 12px;
  }

  a {
    text-decoration: none;
    display: flex;
    align-items: center;
    font-size: 18px;
    transition: 0.2s;
  }

  a:hover {
    color: #c70548;
  }

  :active a {
    color: #fca408;
  }
`;

export default StUserCard;
