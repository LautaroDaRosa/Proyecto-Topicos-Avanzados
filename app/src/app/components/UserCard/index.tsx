import { Link } from 'react-router-dom';
import StUserCard from './StUserCard';
import { MinimalEmployee } from 'types';

const UserCard = ({ id, fullname, photo }: MinimalEmployee) => (
  <StUserCard>
    <Link to={`/profile/${id}`} id={`user-${id}`}>
      <img src={photo} aria-labelledby={`user-${id}`} />
      {fullname}
    </Link>
  </StUserCard>
);

export default UserCard;
