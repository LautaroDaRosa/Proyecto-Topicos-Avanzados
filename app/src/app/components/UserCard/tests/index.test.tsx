import React from 'react';
import { render, screen, within } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import UserCard from '../index';

test('Test render Card', () => {
  const userInfo = {
    fullname: 'Pepocho Tentáculos',
    photo:
      'https://brand.octobot.io/images/Octobot_Mark_Mono_Positive-p-500.jpeg',
    id: 3,
  };

  const { container } = render(
    <Router>
      <UserCard
        id={userInfo.id}
        fullname={userInfo.fullname}
        photo={userInfo.photo}
      />
    </Router>,
  );

  const user = screen.getByRole('link', { name: 'Pepocho Tentáculos' });
  expect(user).toHaveAttribute('href', '/profile/3');
  within(user).getByRole('img', { name: 'Pepocho Tentáculos' });
  // snapshot test
  expect(container).toMatchSnapshot();
});
