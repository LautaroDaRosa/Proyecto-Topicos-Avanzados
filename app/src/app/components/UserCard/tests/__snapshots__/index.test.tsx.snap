// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render Card 1`] = `
.c0 {
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
  -webkit-transition: 0.2s background-color ease-in-out;
  transition: 0.2s background-color ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 img {
  width: 28px;
  border-radius: 50%;
  margin-right: 12px;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 18px;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c0 a:hover {
  color: #c70548;
}

.c0:active a {
  color: #fca408;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/profile/3"
      id="user-3"
    >
      <img
        aria-labelledby="user-3"
        src="https://brand.octobot.io/images/Octobot_Mark_Mono_Positive-p-500.jpeg"
      />
      Pepocho Tent√°culos
    </a>
  </div>
</div>
`;
